akka-http-core/src/main/scala/akka/http/scaladsl/model/headers/headers.scala

```diff
@@ -57,6 +61,60 @@ abstract class CustomHeader extends jm.headers.CustomHeader {
   final def render[R <: Rendering](r: R): r.type = r ~~ name ~~ ':' ~~ ' ' ~~ value
 }
 
+abstract class ModeledCustomHeaderCompanion[H <: ModeledCustomHeader[H]] {
+  def name: String
+  def lowercaseName: String = name.toRootLowerCase
+
+  def parse(value: String): Option[H]
+
+  def apply(value: String): H =
+    parse(value).getOrElse(throw new RuntimeException("Unable to construct custom header by parsing value:" + value))
```

Comment:
I think, as a user, I'd expect a space char here after the `:` in the error message.